src/client.ts

```diff
@@ -66,19 +71,27 @@ class DecisionClient {
   private _api: DecisionApi;
   private _networkId: number;
   private _siteId?: number;
+  private _logger?: LoggerFunc;
 
-  constructor(configuration: Configuration, networkId: number, siteId?: number) {
+  constructor(
+    configuration: Configuration,
+    networkId: number,
+    siteId?: number,
+    logger?: LoggerFunc
```

Comment:
Since this is an internal class and we can guarantee a logger before constructing an instance (either what was passed in or the default), this shouldn't be `nullable`.

Reply:
Updated